spec/GraphQLOverHTTP.md

```diff
@@ -237,120 +311,304 @@ omitted.
+`application/graphql+json` media type.
+
+If the GraphQL response contains the {data} entry and it is not {null}, then the
+server MUST reply with a `2xx` status code and SHOULD reply with `200` status
+code.
 
 Note: The result of executing a GraphQL operation may contain partial data as
 well as encountered errors. Errors that happen during execution of the GraphQL
 operation typically become part of the result, as long as the server is still
-able to produce a well-formed response.
+able to produce a well-formed response. There's currently not an approved HTTP
+status code to use for a "partial response," contenders include WebDAV's status
+code "207 Multi-Status" and using a custom code such as "247 Partial Success."
+[IETF RFC2616 Section 6.1.1](https://datatracker.ietf.org/doc/html/rfc2616#section-6.1.1)
+states "codes are fully defined in section 10" implying that though more codes
+are expected to be supported over time, valid codes must be present in this
+document.
 
-In case of errors that completely prevent the successful execution of the
-request, the server SHOULD respond with the appropriate status code depending on
-the concrete error condition.
+If the GraphQL response contains the {data} entry and it is {null}, then the
+server SHOULD reply with a `2xx` status code and it is RECOMMENDED it replies
+with `200` status code.
 
-The following examples provide guidance on how to deal with specific error
-cases:
+Note: Using `4xx` and `5xx` status codes in this situation is not recommended -
+since no {GraphQL request error} has occurred it is seen as a "partial
+response".
 
-### Unparseable or invalid request body
+If the GraphQL response does not contain the {data} entry then the server MUST
+reply with a `4xx` or `5xx` status code as appropriate.
 
-For example: `NONSENSE`, `{"qeury": "{__typena`
+Note: The GraphQL specification indicates that the only situation in which the
+GraphQL response does not include the {data} entry is one in which the {errors}
+entry is populated.
 
-Completely prevents execution of the GraphQL operation and SHOULD result in
-status code `400` (Bad Request).
+If the GraphQL request is invalid (e.g. it is malformed, or does not pass
+validation) then the server SHOULD reply with `400` status code.
 
-### Document validation
+If the client is not permitted to issue the GraphQL request then the server
+SHOULD reply with `403`, `401` or similar appropriate status code.
 
-Includes validation steps that run before execution of the GraphQL operation:
+#### Examples
 
-- [GraphQL specification validation](https://spec.graphql.org/October2021/#sec-Validation)
-- custom validation, for example: depth limit, complexity limit
+The following examples provide guidance on how to deal with specific error cases
+when using the `application/graphql+json` media type to encode the response
+body:
 
-The server SHOULD deny execution with a status code of `400` (Bad Request).
+##### Invalid request body or parsing failure
```

Comment:
Should this also include failure to parse `variables` or `extensions` provided as GET parameters?

It looks like this is about failures to produce a GraphQL request. But that's separate from GraphQL parse failures... which should probably have its own section?

Reply:
I've tried editing this in, but it just gets more confusing because there's so many different ways of encoding a bad request. Since it's just an example I think it's fine to leave it as-is. What do you think?

Reply:
I think there's a pretty big difference between "you did a GraphQL-over-HTTP-JSON-level error by failing to post good JSON" and "you did a GraphQL-level error by failing to post good GraphQL".  It seems like you are explicitly showing how every other step in the execution process is handled (validation, operation resolution, variable coercion) so not including GraphQL parsing is a notable absence. I'm just talking something like `{"query": "}"}`.

Reply:
Fair. Let's split invalid parameters and parsing failure into two separate examples :+1:  - I'll raise a PR.